torch/csrc/jit/pybind_utils.h

```diff
@@ -187,6 +195,20 @@ inline IValue toIValue(
           return createGenericList(obj, elem_type);
       }
     }
+    case TypeKind::DictType: {
+      const auto& dict_type = type->expect<DictType>();
+      switch (dict_type->getKeyType()->kind()) {
+        case TypeKind::IntType:
```

Comment:
We should centralize the logic for what is a valid key should be somewhere, e.g. in ivalue. Otherwise when it gets expanded, someone will forget to modify all these error checks.